src/cider/nrepl/middleware/debug.clj

```diff
@@ -571,7 +571,9 @@ this map (identified by a key), and will `dissoc` it afterwards."}
                                   (interleave '[dbg break light])
                                   (apply assoc *data-readers*))]
       (try
-        (read-string {:read-cond :allow} code)
+        ;; new-line in REPL always throws; skip for debug convenience
+        (when (> (count code) 3)
```

Comment:
Shouldn't we check for them explicitly? Seems a bit arbitrary to check that something is 3+ characters.

Reply:
I am not completely sure what you mean but we do check for all of them explicitly with that fake reader above. All of them have >= 4 chars this is why 3 characters threshold. It's indeed arbitrary but avoids entering that code in simple cases when there is obviously no reader there. It could be 6 there just as fine :).

As mentioned in the comment, new line in REPL will throw and trigger that catch over there. When debugging I am placing a print there to see exactly what is going on, and those useless throws interfere with the experience. Does this clarify your question?

Reply:
Well, I guess you can just write more explanations here and that would be enough. I just assume that if some code is confusing to me it will probably be confusing to someone else as well.